go-controller/pkg/ovn/pods.go

```diff
@@ -501,6 +502,53 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 		}
 	}
 
+	type Network struct {
+		Name string
+		Ips  []string
+	}
+	routingNamespaceAnnotation := pod.Annotations["k8s.ovn.org/routing-namespace"]
+	if routingNamespaceAnnotation != "" {
+		routingNetworkAnnotation := pod.Annotations["k8s.ovn.org/routing-network"]
+		if routingNetworkAnnotation != "" {
+			var multusNetworks []Network
+			err := json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/network-status"]), &multusNetworks)
+			if err != nil {
+				return fmt.Errorf("unable to unmarshall annotation k8s.v1.cni.cncf.io/network-status on pod %s: %v", pod.Name, err)
+			}
+
+			existingPods, err := oc.watchFactory.GetPods(routingNamespaceAnnotation)
+			if err != nil {
+				return fmt.Errorf("Failed to get all the pods (%v)", err)
+			}
+			for _, multusNetwork := range multusNetworks {
+				if multusNetwork.Name == routingNetworkAnnotation {
+					var gwMask, podMask string
+					for _, gw := range multusNetwork.Ips {
+						if utilnet.IsIPv6(net.ParseIP(gw)) {
```

Comment:
we also need to add these IPs to the nsInfo external gateways object right? so future pods that are created in the namespace get this pod as a GW, also need to handle deletion of the multus pod, and removing the gateway from nsInfo

Reply:
done